app/middlewares/user.js

```diff
@@ -1,5 +1,13 @@
-const { checkIfUserIsLogged: userCheck } = require('./utils');
+const { checkIfUserIsLogged: userCheck } = require('./utils'),
+  { notHasAccessError } = require('../errors');
 
 exports.checkIfUserIsLogged = (req, res, next) => userCheck(req, res, next, false);
 
 exports.checkIfUserIsAdmin = (req, res, next) => userCheck(req, res, next, true);
+
+exports.checkIfUserHasAccess = (req, res, next) => {
+  if (req.params.admin || req.params.user.id.toString() === req.params.id) {
```

Comment:
For more readability:

```javascript
const { admin: isAdmin, user, id } = req.params;

if (isAdmin || String(user.id) === id) return next();
return next(notHasAccessError('You do not have access'));
```

Reply:
- Declare params in order to avoid repeating `req.params` each time.
- `isAdmin` is more representative than `admin`.